AdobeIms/Test/Unit/Model/ConfigTest.php

```diff
@@ -36,28 +30,129 @@ class ConfigTest extends TestCase
+            ->with(Config::XML_PATH_API_KEY);
+
         $this->config->getApiKey();
     }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getPrivateKey
+     */
+    public function testGetPrivateKey(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_PRIVATE_KEY);
+
+        $this->config->getPrivateKey();
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getTokenUrl
+     */
+    public function testGetTokenUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_TOKEN_URL)
+            ->willReturn('URL');
+
+        $this->config->getTokenUrl();
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getAuthUrl
+     */
+    public function testGetAuthUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->exactly(3))
+            ->method('getValue')
+            ->withConsecutive(
+                [Config::XML_PATH_API_KEY],
+                [Custom::XML_PATH_GENERAL_LOCALE_CODE],
+                [Config::XML_PATH_AUTH_URL_PATTERN]
+            )
+            ->willReturn('URL');
+
+        $this->urlMock->expects($this->once())->method('getUrl')->willReturn('URL');
+
+        $this->config->getAuthUrl();
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getCallBackUrl
+     */
+    public function testGetCallBackUrl(): void
+    {
+        $this->urlMock->expects($this->once())
+            ->method('getUrl')
+            ->with('adobe_ims/oauth/callback')
+            ->willReturn('URL');
+
+        $this->config->getCallBackUrl();
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getLogoutUrl
+     */
+    public function testGetLogoutUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_LOGOUT_URL_PATTERN)
+            ->willReturn('URL');
+
+        $this->config->getLogoutUrl('Access token');
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getProfileImageUrl
+     */
+    public function testGetProfileImageUrl(): void
+    {
+        $this->scopeConfigMock->expects($this->exactly(2))
+            ->method('getValue')
+            ->withConsecutive(
+                [Config::XML_PATH_API_KEY],
+                [Config::XML_PATH_IMAGE_URL_PATTERN]
+            )
+            ->willReturn('URL');
+
+        $this->config->getProfileImageUrl();
+    }
+
+    /**
+     * Test for \Magento\AdobeIms\Model\Config::getDefaultProfileImage
+     */
+    public function testGetDefaultProfileImage(): void
+    {
+        $this->scopeConfigMock->expects($this->once())
+            ->method('getValue')
+            ->with(Config::XML_PATH_DEFAULT_PROFILE_IMAGE)
+            ->willReturn('URL');
+
+        $this->config->getDefaultProfileImage();
```

Comment:
Please check that correct value is returned instead of asserting the amount of internal calls